### ISSUE TITLE
FastRTPS v1.10.0 payload data is truncated by 4 bytes. [8152]

### ISSUE THREAD

**Issue by DimaRU** [eProsima/Fast-RTPS#1134]
When reading data using RTPSReader, incoming change data is truncated by 4 bytes. The remote system runs under RTI Connext DDS.
There were no problems with version 1.9.4.

## Expected Behavior

Test code running under FastRTPS v1.9.4:

```
plemu@userv:/mnt/data/rtps94/build/fastrtps/examples/C++/RTPSTest_registered/bin$ ./RTPSTest_registered reader
Starting RTPS example
Registering Reader
Press Enter to stop the Reader.
Received: 48 147112
000000: 00 01 00 00 00 00 00 00 00 00 00 00 01 00 00 00  ................
000010: 00 00 00 00 00 00 00 20 5c 2f 41 40 00 00 00 00  ....... \/A@....
000020: 00 00 00 00 07 00 00 00 6d 73 35 38 33 37 00 00  ........ms5837..
Received: 48 147113
000000: 00 01 00 00 00 00 00 00 00 00 00 00 01 00 00 00  ................
000010: 00 00 00 00 00 00 00 40 33 33 41 40 00 00 00 00  .......@33A@....
000020: 00 00 00 00 07 00 00 00 6d 73 35 38 33 37 00 00  ........ms5837..
Received: 48 147114
000000: 00 01 00 00 00 00 00 00 00 00 00 00 01 00 00 00  ................
000010: 00 00 00 00 00 00 00 80 c2 35 41 40 00 00 00 00  .........5A@....
000020: 00 00 00 00 07 00 00 00 6d 73 35 38 33 37 00 00  ........ms5837..
Received: 48 147115
000000: 00 01 00 00 00 00 00 00 00 00 00 00 01 00 00 00  ................
000010: 00 00 00 00 00 00 00 e0 a3 30 41 40 00 00 00 00  .........0A@....
000020: 00 00 00 00 07 00 00 00 6d 73 35 38 33 37 00 00  ........ms5837..
```
## Current Behavior
Test code running under FastRTPS v1.10.0:

```
plemu@userv:/mnt/data/rtps10/build/fastrtps/examples/C++/RTPSTest_registered/bin$ ./RTPSTest_registered reader
Starting RTPS example
Registering Reader
Press Enter to stop the Reader.
Received: 44 160985
000000: 00 01 00 00 00 00 00 00 00 00 00 00 01 00 00 00  ................
000010: 00 00 00 00 00 00 00 80 eb 51 41 40 00 00 00 00  .........QA@....
000020: 00 00 00 00 07 00 00 00 6d 73 35 38              ........ms58
Received: 44 160986
000000: 00 01 00 00 00 00 00 00 00 00 00 00 01 00 00 00  ................
000010: 00 00 00 00 00 00 00 e0 51 58 41 40 00 00 00 00  ........QXA@....
000020: 00 00 00 00 07 00 00 00 6d 73 35 38              ........ms58
Received: 44 160987
000000: 00 01 00 00 00 00 00 00 00 00 00 00 01 00 00 00  ................
000010: 00 00 00 00 00 00 00 80 14 4e 41 40 00 00 00 00  .........NA@....
000020: 00 00 00 00 07 00 00 00 6d 73 35 38              ........ms58
Received: 44 160988
000000: 00 01 00 00 00 00 00 00 00 00 00 00 01 00 00 00  ................
000010: 00 00 00 00 00 00 00 20 5c 4f 41 40 00 00 00 00  ....... \OA@....
000020: 00 00 00 00 07 00 00 00 6d 73 35 38              ........ms58
```

## Steps to Reproduce
Unfortunately, I can't provide the steps for playback. Writer part is closed-source. Reader part source code is attached.
A trace comparison showed that the 4-byte difference between 1.10.0 and 1.9.4 starts with the **inlineQosSize** value returned by the **updateCacheChangeFromInlineQos** function.

## System information

- Fast-RTPS version: tag: v1.10.0
- OS: Ubuntu 18.04.4
- Network interfaces: eth0 (172.16.1.2)

## Additional resources
- [Reader source code](https://pastebin.com/ZBqJeBf5)
- [sampleData.pcapng.zip](https://github.com/eProsima/Fast-RTPS/files/4451934/sampleData.pcapng.zip)
- Wireshark packet dissect

```
Frame 1: 170 bytes on wire (1360 bits), 170 bytes captured (1360 bits) on interface 0
    Interface id: 0 (en0)
        Interface name: en0
        Interface description: Ethernet
    Encapsulation type: Ethernet (1)
    Arrival Time: Apr  8, 2020 10:52:16.390668000 MSK
    [Time shift for this packet: 0.000000000 seconds]
    Epoch Time: 1586332336.390668000 seconds
    [Time delta from previous captured frame: 0.000000000 seconds]
    [Time delta from previous displayed frame: 0.000000000 seconds]
    [Time since reference or first frame: 0.000000000 seconds]
    Frame Number: 1
    Frame Length: 170 bytes (1360 bits)
    Capture Length: 170 bytes (1360 bits)
    [Frame is marked: False]
    [Frame is ignored: False]
    [Protocols in frame: eth:ethertype:ip:udp:rtps]
    [Coloring Rule Name: UDP]
    [Coloring Rule String: udp]
Ethernet II, Src: Raspberr_cc:5f:8b (b8:27:eb:cc:5f:8b), Dst: AsustekC_a4:75:e7 (40:16:7e:a4:75:e7)
    Destination: AsustekC_a4:75:e7 (40:16:7e:a4:75:e7)
        Address: AsustekC_a4:75:e7 (40:16:7e:a4:75:e7)
        .... ..0. .... .... .... .... = LG bit: Globally unique address (factory default)
        .... ...0 .... .... .... .... = IG bit: Individual address (unicast)
    Source: Raspberr_cc:5f:8b (b8:27:eb:cc:5f:8b)
        Address: Raspberr_cc:5f:8b (b8:27:eb:cc:5f:8b)
        .... ..0. .... .... .... .... = LG bit: Globally unique address (factory default)
        .... ...0 .... .... .... .... = IG bit: Individual address (unicast)
    Type: IPv4 (0x0800)
Internet Protocol Version 4, Src: 172.16.1.26, Dst: 172.16.1.2
    0100 .... = Version: 4
    .... 0101 = Header Length: 20 bytes (5)
    Differentiated Services Field: 0x00 (DSCP: CS0, ECN: Not-ECT)
        0000 00.. = Differentiated Services Codepoint: Default (0)
        .... ..00 = Explicit Congestion Notification: Not ECN-Capable Transport (0)
    Total Length: 156
    Identification: 0xb328 (45864)
    Flags: 0x4000, Don't fragment
        0... .... .... .... = Reserved bit: Not set
        .1.. .... .... .... = Don't fragment: Set
        ..0. .... .... .... = More fragments: Not set
        ...0 0000 0000 0000 = Fragment offset: 0
    Time to live: 64
    Protocol: UDP (17)
    Header checksum: 0x2cec [validation disabled]
    [Header checksum status: Unverified]
    Source: 172.16.1.26
    Destination: 172.16.1.2
User Datagram Protocol, Src Port: 41315, Dst Port: 7413
    Source Port: 41315
    Destination Port: 7413
    Length: 136
    Checksum: 0xbee8 [unverified]
    [Checksum Status: Unverified]
    [Stream index: 0]
    [Timestamps]
        [Time since first frame: 0.000000000 seconds]
        [Time since previous frame: 0.000000000 seconds]
Real-Time Publish-Subscribe Wire Protocol
    Magic: RTPS
    Protocol version: 2.3
        major: 2
        minor: 3
    vendorId: 01.01 (Real-Time Innovations, Inc. - Connext DDS)
    guidPrefix: ac10011a0000006200000001
        hostId: 0xac10011a
        appId: 0x00000062
        instanceId: 0x00000001
    Default port mapping: domainId=0, participantIdx=1, nature=UNICAST_USERTRAFFIC
        [domain_id: 0]
        [participant_idx: 1]
        [traffic_nature: UNICAST_USERTRAFFIC (3)]
    submessageId: INFO_TS (0x09)
        Flags: 0x01, Endianness bit
            0... .... = Reserved: Not set
            .0.. .... = Reserved: Not set
            ..0. .... = Reserved: Not set
            ...0 .... = Reserved: Not set
            .... 0... = Reserved: Not set
            .... .0.. = Reserved: Not set
            .... ..0. = Timestamp flag: Not set
            .... ...1 = Endianness bit: Set
        octetsToNextHeader: 8
        Timestamp: Apr  8, 2020 07:52:16.386869998 UTC
    submessageId: DATA (0x15)
        Flags: 0x07, Data present, Inline QoS, Endianness bit
            0... .... = Reserved: Not set
            .0.. .... = Reserved: Not set
            ..0. .... = Reserved: Not set
            ...0 .... = Reserved: Not set
            .... 0... = Serialized Key: Not set
            .... .1.. = Data present: Set
            .... ..1. = Inline QoS: Set
            .... ...1 = Endianness bit: Set
        octetsToNextHeader: 92
        0000 0000 0000 0000 = Extra flags: 0x0000
        Octets to inline QoS: 16
        readerEntityId: ENTITYID_UNKNOWN (0x00000000)
            readerEntityKey: 0x000000
            readerEntityKind: Application-defined unknown kind (0x00)
        writerEntityId: 0x80008402 (Application-defined writer (with key): 0x800084)
            writerEntityKey: 0x800084
            writerEntityKind: Application-defined writer (with key) (0x02)
        writerSeqNumber: 10762
        inlineQos:
            PID_KEY_HASH
                parameterId: PID_KEY_HASH (0x0070)
                parameterLength: 16
                guid: 33b80eed:cc3aae2f:bd3ac1c6:b7c2073e
            PID_SENTINEL
                parameterId: PID_SENTINEL (0x0001)
        serializedData
            encapsulation kind: CDR_LE (0x0001)
            encapsulation options: 0x0000
            serializedData: 00000000000000000100000000000000000000c0cccc3a40…
```

**Comment by MiguelCompany** [comment]
@DimaRU Please check with #1138, which should fix this issue.

**Comment by MiguelCompany** [comment]
> @DimaRU Please check with #1138, which should fix this issue.

As #1138 has been merged, please check with 1.10.x branch

**Comment by DimaRU** [comment]
I check against 1.10.x branch, now all work as expected. Thank you.  👍🏻
